flutter/flutter
# [flutter_tools] add widget cache for fast reassemble

## Description

Test instructions:

1. flutter config --single-widget-reload-optimization
2. flutter run android/ios/* as normal
3. Perform an edit to the build method of a stateless/stateful widget, observe hot reload time.
4. Perform a second edit to the same build method in the same file, observe improved hot reload time

For consistency, changing the text string in https://github.com/flutter/flutter/blob/master/dev/integration_tests/flutter_gallery/lib/gallery/home.dart#L360 would be a good edit to make
